js_modules/dagster-ui/packages/ui-core/src/assets/useRecentAssetEvents.tsx

```diff
@@ -33,72 +39,109 @@ export function getXAxisForParams(
+    LATEST_ASSET_PARTITIONS_QUERY,
+    {
+      skip: !assetKey,
+      fetchPolicy: 'cache-and-network',
+      variables: {
+        assetKey: {path: assetKey ? assetKey.path : []},
+        limit,
+      },
+    },
+  );
+
+  const {data, loading, refetch} = queryResult;
+  const value = useMemo(() => {
+    const assetNode =
+      data?.assetNodeOrError.__typename === 'AssetNode' ? data?.assetNodeOrError : null;
+    const partitionKeys = assetNode?.partitionKeyConnection?.results || [];
+
+    return {
+      partitionKeys,
+      loading,
+      refetch,
+    };
+  }, [data, loading, refetch]);
+  return value;
+}
+
+export function useRecentAssetEvents(assetKey: AssetKey | undefined, limit: number) {
+  const queryResult = useQuery<RecentAssetEventsQuery, RecentAssetEventsQueryVariables>(
+    RECENT_ASSET_EVENTS_QUERY,
+    {
+      skip: !assetKey,
+      fetchPolicy: 'cache-and-network',
+      variables: {
+        assetKey: {path: assetKey?.path || []},
+        limit,
+      },
+    },
+  );
   const {data, loading, refetch} = queryResult;
 
   const value = useMemo(() => {
     const asset = data?.assetOrError.__typename === 'Asset' ? data?.assetOrError : null;
-    const loaded = {
-      materializations: asset?.assetMaterializations || [],
-      observations: asset?.assetObservations || [],
+    const {materializations, observations} = clipEventsToSharedMinimumTime(
+      asset?.assetMaterializations || [],
+      asset?.assetObservations || [],
+      limit,
+    );
+
+    return {
+      materializations,
+      observations,
+      loading,
+      refetch,
     };
+  }, [data, loading, refetch, limit]);
+
+  return value;
+}
+
+export function useAssetPartitionMaterializations(
+  assetKey: AssetKey | undefined,
+  partitionKeys: string[],
+) {
+  const queryResult = useQuery<AssetPartitionEventsQuery, AssetPartitionEventsQueryVariables>(
+    ASSET_PARTITIONS_MATERIALIZATIONS_QUERY,
+    {
+      skip: !assetKey,
+      fetchPolicy: 'cache-and-network',
+      variables: {
+        assetKey: {path: assetKey ? assetKey.path : []},
+        partitions: partitionKeys,
+      },
+    },
+  );
 
-    const {materializations, observations} = !loadUsingPartitionKeys
-      ? clipEventsToSharedMinimumTime(loaded.materializations, loaded.observations, 100)
-      : loaded;
+  const {data, loading, refetch} = queryResult;
 
-    const allPartitionKeys = asset?.definition?.partitionKeys;
-    const loadedPartitionKeys =
-      loadUsingPartitionKeys && allPartitionKeys
-        ? allPartitionKeys.slice(allPartitionKeys.length - 120)
-        : uniq(
-            [...materializations, ...observations].map((p) => p.partition!).filter(Boolean),
-          ).sort();
+  const value = useMemo(() => {
+    const assetNode =
+      data?.assetNodeOrError.__typename === 'AssetNode' ? data?.assetNodeOrError : null;
+    const materializations = (assetNode?.latestMaterializationByPartition || []).filter((_) => !!_);
```

Comment:
Don't think it's worth changing but if you want, you can also express this as `filter(Boolean)` (running each through the Boolean typecast essentially)